src/ivc/step_folding_circuit.rs

```diff
@@ -428,6 +525,7 @@ where
                     .map(|inp| inp.try_into().unwrap())
             },
         )?;
+        debug!("assigned z_i");
 
         let _assigned_z_next = <Self as StepCircuit<ARITY, C::Base>>::synthesize_step(
```

Comment:
Remove `impl StepCircuit` and  `synthesize_step`.  We don't need StepCircuit for StepFoldingCircuit.

Reply:
As we discussed - not now

Reply:
There's no reason to do that in this already huge PR. I'll do it in a separate PR